python/ray/autoscaler/_private/command_runner.py

```diff
@@ -765,14 +748,45 @@ def run_init(self, *, as_head, file_mounts):
                 for remote, local in cleaned_bind_mounts.items():
                     remote = self._docker_expand_user(remote)
                     if remote not in active_remote_mounts:
-                        cli_logger.error(
-                            "Please ray stop & restart cluster to "
-                            f"allow mount {remote}:{local} to take hold")
+                        try:
+                            is_container_running = not cli_logger.confirm(
+                                False,
+                                "This Docker Container is already Running, Do "
+                                "you want to restart the Docker container on "
+                                "this node?")
+                        except ValueError as e:
+                            cli_logger.print(str(e))
+                        if is_container_running:
+                            cli_logger.error(
+                                "Please ray stop & restart cluster to "
+                                f"allow mount {remote}:{local} to take hold")
             except json.JSONDecodeError:
                 cli_logger.verbose(
                     "Unable to check if file_mounts specified in the YAML "
                     "differ from those on the running container.")
 
+        if not self._check_container_status():
```

Comment:
don't you need to do `if not is_container_running()`?

Reply:
^^ Yep! Thanks for catching this Richard!